File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int result {0};
        unordered_set<int> hash;
        for (int num : nums) {
            hash.insert(num);
        }
        
        for (int i=0; i<nums.size(); i++){
            int count {1};
            int start = nums[i];
            int prev {start - 1};
            int next {start + 1};
            if (hash.find(prev) == hash.end()){
                while (hash.find(next) != hash.end()) {
                    count++;
                    next++;
                }

                if (count > result) result = count;
            }
        }

        return result;

    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        seen = {}
        for num in nums:
            if num in seen:
                seen[num]+= 1
            else:
                seen[num]= 1
        
        return any(count > 1 for count in seen.values())